sys/kern/imgact_aout.c

```diff
@@ -284,13 +286,24 @@ exec_aout_imgact(struct image_params *imgp)
 
 	object = imgp->object;
 	map = &vmspace->vm_map;
+	reservation = virtual_offset;
+	reserv_size = a_out->a_text + a_out->a_data + bss_size;
+
 	vm_map_lock(map);
 	vm_object_reference(object);
 
-	text_end = virtual_offset + a_out->a_text;
+	error = vm_map_reservation_create_locked(map, &reservation,
+	    reserv_size, VM_PROT_RW);
```

Comment:
Hmm, why VM_PROT_RW when this is mapped with a maxprot of RWX below?

Reply:
Because it is a mistake and should be RWX. I did not catch it because we have maxprot in the reservation but `vm_map_insert()` does not enforce them yet in this patch, this is because we would need to remember them as we split the reservation so that when we unmap we return the region to the proper reserved/unmapped state. This requires more refactoring of the reservation identifier, as we currently only use the address of the first reserved entry. Since this patch was getting large enough I deferred it.